lms/envs/common.py

```diff
@@ -589,6 +591,12 @@
 COURSE_ID_PATTERN = COURSE_KEY_PATTERN.replace('course_key_string', 'course_id')
 COURSE_KEY_REGEX = COURSE_KEY_PATTERN.replace('P<course_key_string>', ':')
 
+# Pattern to match a course key or a library key
+COURSELIKE_KEY_PATTERN = r'(?P<course_key_string>({}|{}))'.format(
+    r'[^/:+]+/[^/:+]+/[^/:+]+',
+    r'[^/:]+:[^/+]+\+[^/+]+(\+[^/]+)?',
+)
```

Comment:
I don't like that this changes `COURSELIKE_KEY_PATTERN` without changing `COURSE_KEY_PATTERN`. It seems like the bug (matching the additional slashes after a `library-v1`) would also affect `course-v1` keys that are matched with the `COURSE_KEY_PATTERN`.